### PR TITLE
RSP-1483: Single penalty receipt

### PR DESCRIPTION
**Pull request by jameshighfieldbjss**

### CODE REVIEW

File: test/controllers/receipt.controller.spec.js [hunk 1]
```diff
@@ -99,8 +127,85 @@ describe('ReceiptController', () => {
...
+
+  describe('when PenaltyService and PaymentService both return data for payment code', () => {
+    it('should render the penalty into the receipt view with the payment details', async () => {
+      await singlePaymentReceipt(request, response);
+      const expectedResponse = singlePaymentOkResponse(penaltySvcResp, paymentResp);
+      sinon.assert.calledWith(renderSpy, 'payment/multiPaymentReceipt', expectedResponse);
+    });
+  });
+
+  describe('when PenaltyService rejects on fetching by payment code', () => {
```
Comment (johnmorleybjss) [inline test/controllers/receipt.controller.spec.js]: Couple of blank tests here.

File: src/server/controllers/receipt.controller.js [hunk 2]
```diff
@@ -1,16 +1,79 @@
...
+
+const TIMEZONE_ID = 'Europe/London';
 
 const penaltyGroupService = new PenaltyGroupService(config.penaltyServiceUrl());
+const penaltyService = new PenaltyService(config.penaltyServiceUrl());
 const paymentService = new PaymentService(config.paymentServiceUrl());
 
-export default async (req, res) => {
-  console.log('receipt page');
+function addFormattedPamentDateTime(payment) {
```
Comment (johnmorleybjss) [inline src/server/controllers/receipt.controller.js]: Typo in this function name.

File: src/server/routes.js [hunk 3]
```diff
@@ -16,11 +16,12 @@ router.get('/', mainController.index);
 router.get('/payment-code', paymentCodeController.index);
 router.post('/payment-code', paymentCodeController.validatePaymentCode);
 router.get('/payment-code/:payment_code', paymentCodeController.getPaymentDetails);
+router.get('/payment-code/:payment_code/receipt', singlePaymentReceipt);
```
Comment (johnmorleybjss) [inline src/server/routes.js]: Have you made the change yet to change the redirect URL we send to CPMS to this new route?

Comment (johnmorleybjss) [reply src/server/routes.js]: I believe that redirect URL originates from public portal, so would expect to see it in this PR. Please point out if I've missed it!

Comment (jameshighfieldbjss) [reply src/server/routes.js]: I don't think there's a link between cpms and receipts. There's a link from the penalty details page:

https://github.com/dvsa/rsp-public-portal/blob/849ad9a1b31ac81f16702177437b06db4afc316f/src/server/views/payment/paymentDetails.njk#L71

screen shot 2018-12-13 at 09 40 55

**Comment by jameshighfieldbjss** [comment]
Screenshot of the single penalty receipt:

screen shot 2018-12-13 at 09 48 34

File: src/server/routes.js (hunk 3 above)
Comment (johnmorleybjss) [reply src/server/routes.js]: As discussed, need to determine whether @md23 wants to return to single penalty receipt page after payment or keep the flow as it currently is (I'd imagine it's the former).

File: test/controllers/receipt.controller.spec.js (hunk 1 above)
Comment (jameshighfieldbjss) [reply test/controllers/receipt.controller.spec.js]: I've added the tests for rejection now

File: src/server/controllers/receipt.controller.js (hunk 2 above)
Comment (jameshighfieldbjss) [reply src/server/controllers/receipt.controller.js]: Fixed, thanks.

**Pull request merged**